### ISSUE TITLE
Persons engaged in TheaterEvents

### ISSUE THREAD

**Issue by Harald-Effenberg** [schemaorg/schemaorg#503]
Hi!

Currently there are no Persons allowed in TheaterEvents but attendees, organizers and performers.

But usually the perfomers in a TheaterEvent are actors - why can't we markup them as actors?
In fact, TheaterEvents would be rather boring without actors. And they would in most cases be very chaotic without a director.
Actors and directors are allowed in Movies and TVSeries - why not in TheaterEvents?

And there should be at least an itemprop "contributor" for all the other people who engage themselves for the success of a TheaterEvent: stage designers, costume designers, makeup artists, light engineers, sound engineers...

JM2C
Harry

**Comment by Dataliberate** [comment]
Harry,

Much of what you are asking for is what you would expect from a CreativeWork or subtype —  I notice we have not got a stage based set of CreativeWork subtypes, which might be worth progressing in parallel to TV / Movie subtypes (Play, Show, Musical, etc.) in the way you indicate.

My interpretation of TheaterEvent is that it is mainly about the where/when aspects, with its workPerformed property identify the work(s) that are performed at the event.

I suspect that any proposals you might have for theatre and or live performance oriented enhancements would be received with interest.

~Richard

On 17 May 2015, at 16:14, Harald Effenberg <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi!

Currently there are no Persons allowed in TheaterEvents but attendees, organizers and performers.

But usually the perfomers in a TheaterEvent are actors - why can't we markup them as actors?
In fact, TheaterEvents would be rather boring without actors. And they would in most cases be very chaotic without a director.
Actors and directors are allowed in Movies and TVSeries - why not in TheaterEvents?

And there should be at least an itemprop "contributor" for all the other people who engage themselves for the success of a TheaterEvent: stage designers, costume designers, makeup artists, light engineers, sound engineers...

JM2C
Harry

**Comment by Harald-Effenberg** [comment]
Hi Richard,

thank you for your answer!

I thought about the when/where-thing before I posted this issue, but on the other hand: There are perfomers, aren't they?

And often people go to a TheaterEvent because of the well-known name of a famous director - so this can be an important criterion of the Event, like the names of the performers.

And if there are performers and directors, there should be contributors, too, I think.

Or the performers should be separated from the Event, which then focusses only on the ascpects of time and space, and be only integrated in CreativeWork. They could become Perfomers with - if I could do a wish - a userfriendly itemprop="characterName" without the necessity to be embedded within a PerformanceRole (which neither Google nor Bing seem to be able to process correctly).

Kind regards
Harry

**Comment by danbri** [comment]
Allowing 'actor', 'director', 'contributor' (e.g. with a Role) on these kinds of Event (i.e. TheaterEvent) sounds plausible and useful to me. Perhaps 'contributor' could be attached at a higher level, e.g. to Event. At some point we'll be asked about Action too, I'm sure; but let's ignore those for now. 

Since we have FestivalEvent, VisualArtsEvent, LiteraryEvent, PublicationEvent my inclination is to attach actor/director/contributor at the Event level. @vholland any thoughts?

**Comment by vholland** [comment]
It sounds useful to me. It might be nice to consider all of the changes to Event together. Quickly skimming, that would include issue #447, issue #445, and issue #406.

**Comment by WeaverStever** [comment]
Looking at the properties for https://schema.org/director, I see that https://schema.org/play is not among the expected types.

Being an event, "TheaterEvent" is not useful n the context of a collection. (A play can run for years on Broadway). "Play" should be included as a Type in context similar to 'Clip" or "Movie"

In this way. the requirement for dateCreated can be satisfied simply by year and and the extraneous Event (required) properties are not needed.


This example passes SDTT, "Play" and probably "Podcast" should probably be added as expected types.

~~~~
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Movie",
  "actor": [
    {
      "@type": "Person",
      "name": "Johnny Depp"
    },
    {
      "@type": "Person",
      "name": "Penelope Cruz"
    },
    {
      "@type": "Person",
      "name": "Ian McShane"
    }
  ],
  "aggregateRating": {
    "@type": "AggregateRating",
    "bestRating": "10",
    "ratingCount": "200",
    "ratingValue": "8",
    "reviewCount": "50"
  },
  "author": [
    {
      "@type": "Person",
      "name": "Ted Elliott"
    },
    {
      "@type": "Person",
      "name": "Terry Rossio"
    }
  ],
  "description": "Jack Sparrow and Barbossa embark on a quest to find the elusive fountain of youth, only to discover that Blackbeard and his daughter are after it too.",
  "director": {
    "@type": "Person",
    "name": "Rob Marshall"
  },
  "name": "Pirates of the Carribean: On Stranger Tides (2011)",
  "image": "https://example.com/poster.jpg",
  "dateCreated": "2011"
}
</script>

~~~~
 
 If I change the type on the script above to Play, Director, Actor and itemReviewed fails the SDTT.

**Comment by philbarker** [comment]
@WeaverStever I disagree. A play only gets a director when it is performed and the definition of schema.org/Play makes it clear that "the peformance of a Play would be a TheaterEvent".  Different productions of a play will have different directors (indeed a long running Broadway production may have several). It makes no sense to list all the directors of Othello since 1603 against the same entity. The same is not true of film and broadcast media as these are "captured" performances in part created by the director.  

SDTT requirements are, of course, distinct from schema.org requirements.

**Comment by WeaverStever** [comment]
@philbarker

A Play (singular) is an Event a Play (plural) can also be a collection of Events -- same word, different context.  In addition, a Play has a run, off Broadway, on Broadway etc. with different casts and directors.  

My use case is for an actor's resume, because she played 102 performances, twenty years ago, there is no "Offers" to enumerate and the Location/address are completely extraneous, because she played both the off-Broadway and Broadway performances. Additionally.I can't think of any play that did not have a director.

The Play type is pending, and I do note the typo in the word "peformance", the passage should read, "**A performance** of a play would be a TheaterEvent"

As a collection, the Play (or run) is a different context for the same word. If I can't conform to the Rich Snippets (SDTT) definition, I am better off not including the structured data -- because the website gets dinged for performance problems.

**Comment by RichardWallis** [comment]
Taking the definition from the term "_A play is a form of literature, usually consisting of dialogue between characters, intended for theatrical performance_" and also referencing a dictionary definition I am aware of "a dramatic work for the stage or to be broadcast" and putting it together with a tweak to the note in the description "**A performance** of a play would be a TheaterEvent **BroadcastEvent**", It is clear to me that a play is the work performed at an event, which would have associated actors, directors etc.

In describing a run of events sharing the same workPerformed, actors etc. I would look to using the [EventSeries](https://schema.org/EventSeries) type.

**Comment by RichardWallis** [comment]
Tweak to comment, fixing typo, applied directly to source:
"A performance of a play would be a TheaterEvent or BroadcastEvent"

**Comment by MarieCooperActor** [comment]
I'm having the same issue trying to add plays I have acted in, in the past, but I have just had to leave "offers" in the GSC warnings.

I tried to guesstimate the old ticket price just to get the  'offers' warning to go away, but it opens up a whole new can of availability, price currency, URL and validFrom  whoopass.

**Comment by WeaverStever** [comment]
@RichardWallis

Unfortunately, Google Rich Snippets seems to understand Event(s) as pending performance.   Using the Event structure completely overcomplicates associating a person with a performance(s) of a work.  Currency, price, doortime etc are all irrelevant after the event has passed.

IMHO Play should have the same properties as Movie.

**Comment by WeaverStever** [comment]
Also, in the English language, the word "Play" could mean the actual written work, an event, or a collection of performances at one or more locations.

**Comment by WeaverStever** [comment]
Similar to https://schema.org/Movie, the following have specific properties like actor, director and musicby

https://schema.org/Clip
https://schema.org/CreativeWorkSeason
https://schema.org/Episode
https://schema.org/MovieSeries
https://schema.org/TVSeries

I'm failing to see a reason why https://schema.org/Play does not follow the same convention.

**Comment by philbarker** [comment]
The difference between Movie and Play arises because remakes of movies are considered different movies but different productions of a play are still considered productions of the same play. So you can list the cast of The Parent Trap (1961 film) and The Parent Trap (1998 film) without ambiguity, but you cannot list the cast of Othello without saying which production. Natural language may be rather imprecise in using "play" for both the creative work and the production, but schema.org isn't.

You might have better luck raising problems that arise from Google rich snippets and Google search console in the appropriate Google support forums.

**Comment by WeaverStever** [comment]
@philbarker No, each time a Play travels to a new location it can be considered a production, i.e., a new CreativeWorkSeries.  The set is relocated, lighting and sound equipment installed and the cast sometimes varies as may the musicians. (We have no sub-definition for Musical or Opera.)

You've just disambiguated two performances of an identical work  (_The Parent Trap_) by year, further disambiguation for a Play can be accomplished by location, language adaptation (a new work by the same name),  director and production company etc.  Some plays have more than one production unit, so the play could be performed in multiple locations on the same night.

In music, we have creativeworks where the work and the performance are named the same, these are further disambiguated by recording and release -- often the same title and often distributed on different media types.   The recordings can be included on other albums (greatest hits), or the work can be licensed as cover songs to other artists for new recordings and releases.  The schema does a pretty good job of covering these nuances, yet the properties for Play are generic Creativework and generic Thing.

This problem is not with Google, it is with an incomplete instantiation of the type Play.  

These two things are not the same

Thing > CreativeWork > Play
Thing > Event > TheaterEvent

The event is just one of many in the collection, yet we currently have no way to disambiguate the collection(s).

([Theatrical definitions](https://www.thefreedictionary.com/Theatre+play)) Play
n.
1.
a. A literary work written for performance on the stage; a drama.
b. The performance of such a work.

v.tr.
2.
a. To perform or act (a role or part) in a dramatic performance.
b. To assume the role of; act as: played the peacemaker at the meeting.
c. To pretend to be; mimic the activities of: played cowboy; played the star.

3.
a. To perform (a theatrical work or part of a work): The actors played the scene with great skill.
b. **To present a theatrical performance or other entertainment in (a given place): The company played Boston last week.**

What we are missing is the ability to use Play in [CreativeWorkSeries](https://schema.org/CreativeWorkSeries).  To define the collection of events/performances between the production's opening night and grand finale performance -- given other specific properties like location, production company, director etc.

See also https://github.com/schemaorg/schemaorg/issues/1816
.

**Comment by WeaverStever** [comment]
@RichardWallis I appreciate the pointer to the EventSeries type, but again, the Event type is becoming to be understood (SDTT and Rich Snippets) as a [pending performance](https://developers.google.com/search/docs/data-types/event). (Covid-19 changed the way Google handles Events)

For instance, EventStatus and ItemAvailability have no past-tense enumeration members.

**EventStatus Enumeration members**

- EventCancelled
- EventMovedOnline
- EventPostponed
- EventRescheduled
- EventScheduled

 
**ItemAvailability Enumeration members**

- BackOrder
- Discontinued
- InStock
- InStoreOnly
- LimitedAvailability
- OnlineOnly
- OutOfStock
- PreOrder
- PreSale
- SoldOut

The need here is to be able to describe a Play or collection of Plays that have already been performed (for the purpose of a resume or CV) without having to name every location, ticket price and currency.  In fact, most ticketing agencies remove the Event information from their websites on the day after the performance. (and even they can't seem to get the schema correct/complete)

**Comment by RichardWallis** [comment]
The capability of [Event](https://schema.org/Event) to be used for the description of events from any time past or present has always been there.  The clarification of that was the subject of issue #1931 which resulted in a proposal for the term's description to include "_An event happening at a certain time past or future and location physical or virtual..._".  This moved on to become PR #2660 which for some reason stalled.

The [validator](https://validator.schema.org/) already passes the example associated with the PR:
```
         "@context": "http://schema.org",
	  "@type": "Event",
	  "name": "Signing of the Magna Carta",
	  "startDate": "1215-06-15",
	  "location": {
	    "@type": "Place",
	    "name": "Runnymede",
	    "address": {
	      "@type": "PostalAddress",
	      "addressLocality": "Egham",
	      "addressRegion": "Surrey",
	      "postalCode": "SL4 2JL",
	      "streetAddress": "Windsor Road",
	      "addressCountry": "UK"
	    },
	    "sameAs": "https://en.wikipedia.org/wiki/Runnymede"
	  },
	  "contributor": {
	     "@type": "Person",
	     "name": "King John of England",
	     "sameAs": "http://www.wikidata.org/entity/Q129308"
	  },
	  "sameAs": "http://www.wikidata.org/entity/Q12519"
	}
```
With focus of search engines I am not surprised that it may not qualify for a rich results feature.  

As there are at least as many notable events in the past as there will be in the near future, I believe that the emphasising of the historical capabilities of Event should be revisited, starting with PR #2660. If nothing else, this is of significant relevance to the cultural heritage community.

To the examples provided. Like all properties _EventStatus_ is optional, although a possible enumeration addition of _EventPast_ could be considered.  I'm not clear as to he relevance of _ItemAvailability_ to Event description.

**Comment by philbarker** [comment]
I don't think _EventPast_ is necessary because Event has the endDate property.  All the current items in _EventStatus_ work for past events.

aside: @RichardWallis I think the definition of `schema:sameAs` means the example should use `"sameAs": "http://www.wikidata.org/wiki/Q129308"` to point to the wikidata webpage rather than the `/entity/` identifier, though it's too small a hill to die on.

**Comment by RichardWallis** [comment]
@philbarker I agree with you on _EventPast_. 

```/wiki/``` or ```/entity/``` - one of those discussions that will rumble on for years - but I agree on the size of the hill.

**Comment by danbri** [comment]
Can this discussion be grounded in an application that consumes the data?

**Comment by WeaverStever** [comment]
@RichardWallis, the Event example for "Signing of the Magna Carta" generates the following warnings on SDTT.

description	
endDate	
eventAttendanceMode	
eventStatus	
image	
offers	
organizer

Additionally, in the WikiData item, Magna Carta is not described as an event, the instances are charter, contract and manuscript.  To add event to the Magna Carta instances, the following are available:

**occurrence (event)**
occurrence of a fact or object in space-time; instantiation of a property in an object

**event**
temporary and scheduled happening, like a conference, festival or competition

It appears that we (schema) have a faulty understanding of the term Event and we are trying to shoehorn past-events into the type because the properties happen to be the same. IMHO the "Signing of the Magna Carta" was not a "temporary and scheduled happening, like a conference, festival or competition".

@danbri mentions the consumer of the data, I contend that Google, in addition to being this project's sponsor, it the largest consumer of schema data.  

From the eventStatusType on the Google guidance for Event:
_The status of the event. **If you don't use this field, Google understands the eventStatus to be EventScheduled.** Here are the supported values. You can use multiple statuses, if applicable (for example, if an event was both rescheduled and moved online)._  https://developers.google.com/search/docs/data-types/event

Both Google and WikiData appear to understand "event" to be an instance of a pending occurrence.
